File: assignments/week1/day1/problem3.java
// iii) Find the largest number less than N whose each digit is prime number

// intuition
// one digit prime numbers -> 2, 3, 5, 7

import java.util.Scanner;

class Solution{
    public static void main(String[] args){
        Scanner scn = new Scanner(System.in);
        int num = scn.nextInt();

        for(int n = num-1; n>1; n--){
            if(isEachDigitPrime(n)){
                System.out.println(n);
                return;
            }
        }

        System.out.println("not found");
    }

    /*
     * MARK :- Check if each digit of given num is prime or not
     * INPUT :- int Number
     * OUTPUT :- boolean
     * DESCRIPTION :- Check if each digit of given num is prime or not. It check for only 2,3,5 and 7 as we have to check a digit of a num;
     */
    public static boolean isEachDigitPrime(int num){
        while(num>0){
            int dig = num % 10;
            num = num / 10;
            
            if(dig != 2 && dig != 3 && dig != 5 && dig != 7){
                return false;
            }
        }

        return true;
    }
}

File: assignments/week1/day1/problem3.py
# // iii) Find the largest number less than N whose each digit is prime number

def largest_prime(num:int):
    for n in range(num,0, -1):
        if is_each_digit_prime(n):
            return n
        
    return -1

def is_each_digit_prime(num:int):
    while num > 0:
        dig = num % 10
        num = num // 10

        if dig != 2 and dig != 3 and dig != 5 and dig != 7:
            return False
    
    return True


print(largest_prime(770))